### PR TITLE
Safety features

### PR DESCRIPTION
**Pull request by AaronMcGarry**
Various non-necessary pieces of code that affect how the car drives

### CODE REVIEW

File: Gobabygo/Gobabygo.ino [hunk 1]
```diff
@@ -2,58 +2,84 @@ const int joyStickXPin (0);
...
+  timeSincePrint += currentTime - previousTime;
+  timeSinceAcceleration = currentTime - previousTime;
   
   joyXVal = analogRead(joyStickXPin);
-  scaledJoyXVal = joyXVal/1023*2 - 1;
-  correctForDeadzone(scaledJoyXVal, deadzone);
-  
-  // ledVal = map(joyXVal, 0, 1023, 0, 255);
-  // analogWrite(rightMotorPin, ledVal);
+  joyXVal = 2/1023 * joyXVal - 1;
```
Comment (pordonj) [inline Gobabygo/Gobabygo.ino:62]: did this work in the code this is replacing? `2/1023` immediately jumps out to me - unless this works fine on a pi?

File: Gobabygo/Gobabygo.ino [hunk 2]
```diff
@@ -31,25 +31,27 @@ void setup() {
...
+  int joyXVal;
   float scaledJoyXVal;
   int joyYVal;
   float scaledJoyYVal;
   
   float rightMotorVal;
   float leftMotorVal;
   float scaler;
+
+  const float backwardPower = 0.4;
```
Comment (jdlanning) [inline Gobabygo/Gobabygo.ino]: Suggest to declare this above with other constants like dead zone

Comment (jdlanning) [reply Gobabygo/Gobabygo.ino]: We can think of the top of the file as being same as our Constants.java. Having them above makes it easy to tune/ tweak them easily.

File: Gobabygo/Gobabygo.ino [hunk 3]
```diff
@@ -2,58 +2,84 @@ const int joyStickXPin (0);
...
   float leftMotorVal;
   float scaler;
+
+  int timeSincePrint;
+  int timeSinceAcceleration;
+
+  const float deadzone = 0.1;
+  const float backwardPower = 0.4;
+  const float turnScaler = 0.5;
+  const float accelerationPerMs = 0.0005;
```
Comment (jdlanning) [inline Gobabygo/Gobabygo.ino]: Same with these constants, suggest to declare at top of file for ease of modification

File: Gobabygo/Gobabygo.ino [hunk 4]
```diff
@@ -2,58 +2,84 @@ const int joyStickXPin (0);
...
-  float scaledJoyXVal;
-  int joyYVal;
-  float scaledJoyYVal;
-  
   float rightMotorVal;
   float leftMotorVal;
   float scaler;
+
+  int timeSincePrint;
+  int timeSinceAcceleration;
```
Comment (jdlanning) [inline Gobabygo/Gobabygo.ino]: It’s best to initialize these values to 0 for clarity. Arduino may have default initialization but not all programming languages do so it’s a good habit to define an initial value

**Review by jdlanning — COMMENTED** [review]
Nice work on these changes @AaronMcGarry , really like the thought you put into the usability / driving experience of the kids who will use the cars controlled by this code. The slow reverse, careful turning at high speeds and the acceleration limit will all help give a better experience to the end user.

Just a few comments to address.

Also very sorry on the delayed review, hope you are enjoying your summer.

File: Gobabygo/Gobabygo.ino (hunk 1 above)
Comment (AaronMcGarry) [reply Gobabygo/Gobabygo.ino:62]: They both worked, I just changed it so it would be formatted the same for both joystick variables.

**Review by jdlanning — APPROVED** [review]

**Pull request merged**